step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java

```diff
@@ -0,0 +1,104 @@
+package step.plugins.timeseries.dashboards.model;
+
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+public class ChartFilterItem {
+	
+	private String label; // optional. should be used for filter options
```

Comment:
Seems this class is having optional fields depending on the filter type, shouldn't we have a class hierarchy here?

Reply:
@david-stephan all the attributes are together now. We can split into multiple classes based on the type, but I'm not sure it is worth it. This way the processing is very straight in both FE and BE.